src/formatDate.js

```diff
@@ -9,6 +9,41 @@
  */
 function formatDate(date, fromFormat, toFormat) {
   // write code here
+  const oldSeparator = fromFormat.pop();
+  const newSeparator = toFormat.pop();
+
+  const splitedDate = date.split(oldSeparator);
+
+  const newDate = [];
+
+  let yearFormat = toFormat.includes('YY') ? 'YY' : 'YYYY';
+
+  for (let i = 0; i < splitedDate.length; i++) {
+    const oldPart = fromFormat[i];
+    let index = toFormat.indexOf(oldPart);
+
+    if(oldPart.includes('Y') && oldPart !== yearFormat) {
+      index = toFormat.indexOf(yearFormat);
+    }
+
+    const newPart = toFormat[index];
+    
+    if (oldPart.length > newPart.length) {
+      splitedDate[i] = splitedDate[i].split('').slice(-2).join('');
+    }
+
+    if (oldPart.length < newPart.length) {
+      splitedDate[i] =
+        +splitedDate[i] < 30
+          ? `20${splitedDate[i]}`
+          : `19${splitedDate[i]}`;
+    }
+
+    newDate[index] = splitedDate[i];
+
+  }
```

Comment:
The loop iterates over the `splitedDate` array and attempts to reformat and reorder the date parts. However, the logic for reordering is incorrect. You should map the parts of the `fromFormat` to their corresponding indexes in the `toFormat`. The current implementation assumes that the order and presence of parts are the same in both formats, which may not be the case.